test/cmd/apply.sh

```diff
@@ -191,8 +190,7 @@ __EOF__
   # check right pod exists
   kube::test::get_object_assert 'pods a' "{{${id_field:?}}}" 'a'
   # check wrong pod doesn't exist
-  output_message=$(! kubectl get pods b -n nsb 2>&1 "${kube_flags[@]:?}")
-  kube::test::if_has_string "${output_message}" 'pods "b" not found'
+  kube::test::wait_object_assert 'pods -n nsb' "{{range.items}}{{${id_field:?}}}:{{end}}" ''
```

Comment:
I'm confused why this was looking in namespace nsb... the apply on line 188 is not for that namespace

Reply:
I think that's because https://github.com/kubernetes/kubernetes/blob/master/hack/testdata/prune/b.yaml specify the namespace, while https://github.com/kubernetes/kubernetes/blob/master/hack/testdata/prune/a.yaml not.

Reply:
but we're not applying prune/b.yaml between line 182 and here

Reply:
Hmm I am not the author of this test, perhaps he wants to make sure pod b does not exist before applying b.yaml?